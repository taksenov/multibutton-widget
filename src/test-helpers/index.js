function checkFunc(dom, selector) {
  if (typeof dom.update === 'function') {
    const el = dom.update().find(selector);
    if (el.exists()) {
      return el;
    }
    return null;
  }
  const els = dom.querySelectorAll(selector);
  if (els.length !== 0) {
    return els;
  }
  return null;
}


export async function waitForSelection(dom, selector) {
  let numSleep = 0;
  for (;;) {
    const el = checkFunc(dom, selector);
    if (el) {
      return el;
    }
    if (numSleep > 2) {
      throw new Error(`could not find ${selector}`);
    }
    await new Promise(resolve => setTimeout(resolve, 250));
    numSleep += 1;
  }
}
