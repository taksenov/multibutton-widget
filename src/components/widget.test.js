import React from 'react';
import { mount } from 'enzyme';
import Widget from './widget';
import { waitForSelection } from '../test-helpers';

describe('<Widget />', () => {
  test('open/close', async () => {
    const widgetDom = mount(<Widget />);
    expect(widgetDom).toMatchSnapshot();

    {
      const dockAnchorEl = widgetDom.find('a.dock');
      expect(dockAnchorEl).toHaveLength(1);
      // open widget
      dockAnchorEl.simulate('click');
    }

    expect(widgetDom).toMatchSnapshot();

    // dock does not exist anymore
    expect(widgetDom.find('a.dock')).toHaveLength(0);

    const closeAnchorEl = await waitForSelection(widgetDom, 'a.widget-header-icon');

    expect(closeAnchorEl).toHaveLength(1);
    // close widget
    closeAnchorEl.simulate('click');

    {
      const dockAnchorEl = await waitForSelection(widgetDom, 'a.dock');
     expect(dockAnchorEl).toHaveLength(1);
    }

    expect(widgetDom).toMatchSnapshot();
  });

});
