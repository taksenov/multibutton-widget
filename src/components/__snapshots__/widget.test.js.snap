// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Widget /> open/close 1`] = `
<Widget>
  <div
    className="docked-widget"
    style={
      Object {
        "right": "20px",
      }
    }
  >
    <Transition
      appear={false}
      enter={true}
      exit={true}
      in={false}
      mountOnEnter={false}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      timeout={250}
      unmountOnExit={false}
    >
      <div
        className="widget widget-exited"
      >
        <ButtonsList
          left={false}
          onButtonClick={[Function]}
          parsers={Array []}
        >
          <div />
        </ButtonsList>
        <MainButton
          altText="Close"
          backgroundColor="#969696"
          handleToggleOpen={[Function]}
          image="close.svg"
          left={false}
        >
          <div
            className="buttonDock"
            style={
              Object {
                "justifyContent": "flex-end",
              }
            }
          >
            <div
              className="buttonWrapper"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#969696",
                }
              }
            >
              <img
                alt="Close"
                height={50}
                src="close.svg"
                width={50}
              />
            </div>
          </div>
        </MainButton>
      </div>
    </Transition>
    <MainButton
      altText="sliza.ru"
      backgroundColor="#DD9021"
      handleToggleOpen={[Function]}
      image="slizaDock.svg"
      left={false}
    >
      <div
        className="buttonDock"
        style={
          Object {
            "justifyContent": "flex-end",
          }
        }
      >
        <div
          className="buttonWrapper"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#DD9021",
            }
          }
        >
          <img
            alt="sliza.ru"
            height={50}
            src="slizaDock.svg"
            width={50}
          />
        </div>
      </div>
    </MainButton>
  </div>
</Widget>
`;

exports[`<Widget /> open/close 2`] = `
<Widget>
  <div
    className="docked-widget"
  >
    <Transition
      appear={false}
      enter={true}
      exit={true}
      in={true}
      mountOnEnter={false}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      timeout={250}
      unmountOnExit={false}
    >
      <div
        className="widget widget-entering"
      >
        <div
          className="widget-header"
        >
          <div
            className="widget-header-title"
          >
            Header
          </div>
          <a
            className="widget-header-icon"
            onClick={[Function]}
          >
            X
          </a>
        </div>
        <div
          className="widget-body"
        >
          Body
        </div>
        <div
          className="widget-footer"
        >
          Footer
        </div>
      </div>
    </Transition>
  </div>
</Widget>
`;

exports[`<Widget /> open/close 3`] = `
<Widget>
  <div
    className="docked-widget"
  >
    <Transition
      appear={false}
      enter={true}
      exit={true}
      in={false}
      mountOnEnter={false}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      timeout={250}
      unmountOnExit={false}
    >
      <div
        className="widget widget-exited"
      >
        <div
          className="widget-header"
        >
          <div
            className="widget-header-title"
          >
            Header
          </div>
          <a
            className="widget-header-icon"
            onClick={[Function]}
          >
            X
          </a>
        </div>
        <div
          className="widget-body"
        >
          Body
        </div>
        <div
          className="widget-footer"
        >
          Footer
        </div>
      </div>
    </Transition>
    <a
      className="dock"
      onClick={[Function]}
    >
      ^ OPEN ^
    </a>
  </div>
</Widget>
`;
